backend/libinput/pointer.c

```diff
@@ -141,6 +141,54 @@ void handle_pointer_axis(struct libinput_event *event,
 	wlr_signal_emit_safe(&wlr_dev->pointer->events.frame, wlr_dev->pointer);
 }
 
+void handle_pointer_axis_wheel(struct libinput_event *event,
+		struct libinput_device *libinput_dev) {
+	struct wlr_input_device *wlr_dev =
+		get_appropriate_device(WLR_INPUT_DEVICE_POINTER, libinput_dev);
+	if (!wlr_dev) {
+		wlr_log(WLR_DEBUG, "Got a pointer event for a device with no pointers?");
+		return;
+	}
+	struct libinput_event_pointer *pevent =
+		libinput_event_get_pointer_event(event);
+	struct wlr_event_pointer_axis wlr_event = { 0 };
+	wlr_event.device = wlr_dev;
+	wlr_event.time_msec =
+		usec_to_msec(libinput_event_pointer_get_time_usec(pevent));
+
+	if (libinput_event_pointer_get_axis_source(pevent) != LIBINPUT_POINTER_AXIS_SOURCE_WHEEL) {
+		wlr_log(WLR_DEBUG, "Got a wheel axis event for a non-wheel device");
+		return;
+	}
+	wlr_event.source = WLR_AXIS_SOURCE_WHEEL;
+
+	const enum libinput_pointer_axis axes[] = {
+		LIBINPUT_POINTER_AXIS_SCROLL_VERTICAL,
+		LIBINPUT_POINTER_AXIS_SCROLL_HORIZONTAL,
+	};
+	for (size_t i = 0; i < sizeof(axes) / sizeof(axes[0]); ++i) {
```

Comment:
you don't have an `ARRAY_SIZE` or a macro like that? do you really open-code this everywhere?

Reply:
We don't use macros.